api/desecapi/pdns.py

```diff
@@ -103,3 +105,31 @@ def get_rrset_datas(domain):
              'records': [record['content'] for record in rrset['records']],
              'ttl': rrset['ttl']}
             for rrset in get_zone(domain)['rrsets']]
+
+
+def construct_catalog_rrset(zone=None, delete=False, subname=None, qtype='PTR', rdata=None):
+    # subname can be generated from zone for convenience; exactly one needs to be given
+    assert (zone is None or subname is None)
+    assert not (zone is None and subname is None)
```

Comment:
alternative: assert (a is None) ^ (b is None)

Reply:
nice! fixed